### ISSUE TITLE
Accessibility: NavigationView items are missing containers.

### ISSUE THREAD

**Issue by MarcoZehe** [microsoft/Xaml-Controls-Gallery#658]
**Describe the bug**

A group of navigationView items should have a semantic accessibility container. If set to left vertical, items are list items, but are missing a listbox container. If set to top horizontal, they are tabs, but are missing a tab list container. Without these, screen readers like Narrator or NVDA don't tell the user what navigation paradigm to use for arrow key navigation.

**Steps to reproduce the bug**

1. Go to the NavigationView example page.
2. Turn on Narrator and tab through the examples.

**Expected behavior**

Narrator should speak "list" or "listbox" followed by the item in the vertical example, and it should possibly speak "tabs" or "tab list" for the horizontal top orientation examples. The latter are more clear since tab controls are announced as real roles by screen readers, but list items usually are not.

**Actual behavior**

Narrator only speaks the item count and the name, but not what these are for the vertical examples. For the horizontal ones, it at least indicates that these are tabs.

**Screenshots**

**Version Info**
- [x] App downloaded from Store 
    - If so, which version? (Found on the Settings page of the app.) Version: 1.2.17.0
- [ ] App built locally from source 
  - If so, which branch? (WinUI 3 versions are built off the winui3preview branch)

**Additional context**

Windows 10 version:
- [ ] Insider Build (xxxxx)
- [x] October 2020 Update (19042)
- [ ] May 2020 Update (19041)
- [ ] November 2019 Update (18363)
- [ ] May 2019 Update (18362)
- [ ] October 2018 Update (17763)
- [ ] April 2018 Update (17134)
- [ ] Fall Creators Update (16299)
- [ ] Creators Update (15063)
- [ ] Anniversary Update (14393)

Device form factor:
- [x] Desktop
- [ ] Mobile
- [ ] Xbox
- [ ] Surface Hub
- [ ] IoT

**Comment by chingucoding** [comment]
@YuliKl FYI
